readthedocs/projects/tasks.py

```diff
@@ -743,15 +744,23 @@ def fileify(version_pk, commit):
                 if fnmatch.fnmatch(filename, '*.html'):
                     dirpath = os.path.join(root.replace(path, '').lstrip('/'),
                                            filename.lstrip('/'))
-                    obj, created = ImportedFile.objects.get_or_create(
-                        project=project,
-                        version=version,
-                        path=dirpath,
-                        name=filename,
-                        commit=commit,
-                    )
-                    if not created:
-                        obj.save()
+                    full_path = os.path.join(root, filename)
+                    md5 = hashlib.md5(open(full_path, 'rb').read()).hexdigest()
+                    try:
+                        obj, created = ImportedFile.objects.get_or_create(
+                            project=project,
+                            version=version,
+                            path=dirpath,
+                            name=filename,
+                        )
+                    except Exception:
```

Comment:
IMO exception catches should be as specific as possible. Are you expecting any database errors or similar here? The log message also won't give any hints about what actually went wrong. If you really need to catch the very generic `Exception`, then at least document why you are catching ALL exceptions here.